apps/status-api/src/controllers/OracleStatusController.ts

```diff
@@ -0,0 +1,43 @@
+import { Controller, Get, Query } from '@nestjs/common'
+import { WhaleApiClient } from '@defichain/whale-api-client'
+import { OraclePriceFeed } from '@defichain/whale-api-client/dist/api/Oracles'
+import { SemaphoreCache } from '../../../whale/src/module.api/cache/semaphore.cache'
+
+@Controller('oracles')
+export class OracleStatusController {
+  constructor (
+    private readonly client: WhaleApiClient,
+    protected readonly cache: SemaphoreCache
+  ) {
+  }
+
+  @Get()
+  async getOracleStatus (
+    @Query('address') oracleAddress: string
+  ): Promise<{ [key: string]: string }> {
+    const oraclePriceFeed: OraclePriceFeed = await this.cachedGet(`oracle-${oracleAddress}`, async () => {
+      const oracle = await this.client.oracles.getOracleByAddress(oracleAddress)
+      return (await this.client.oracles.getPriceFeed(oracle.id, oracle.priceFeeds[0].token, oracle.priceFeeds[0].currency, 1))[0]
+    }, 5000)
+
+    const nowEpoch = Date.now()
+    const latestPublishedTime = oraclePriceFeed.block.medianTime * 1000
+    const timeDiff = nowEpoch - latestPublishedTime
+
+    return {
+      status: timeDiff <= (45 * 60 * 1000) ? 'operational' : 'outage'
```

Comment:
How about ticker "holiday"?

Reply:
Hmm how would we determine if the ticker is on 'holiday'?

Reply:
Probably is too much to be able to precisely determine holiday. Perhaps can check for weekend (might have to make assumption for ticker time zone).

Any thoughts for not showing `outage` when it is not meant to? @fuxingloh @joeldavidw

Reply:
The providers will be repeatedly return with the last known price when the market closes over the weekends to keepalive. So in the event that there are no published time, it will be an outage.

Reply:
😅 then can ignore the question in the first place, the source is up all time regardless "holiday"